pinc/JsonStorage.inc

```diff
@@ -0,0 +1,91 @@
+<?php
+
+class JsonStorage
+{
+    private string $username;
+
+    public function __construct(string $username)
+    {
+        $this->username = $username;
+    }
+
+    // -------------------------------------------------------------------------
+
+    /**
+     * Set or update a json_storage object.
+     */
+    public function set(string $setting, string $value): void
+    {
+        // It's possible $value could be very large and including it twice
+        // in the query will cause it to be too big. The default max query
+        // size is 64MB which means this will fail if the JSON blob is
+        // just shy of 32MB. That seems plenty big.
+        $sql = sprintf(
+            "
+            INSERT INTO json_storage
```

Comment:
Does this amount to the same thing as 'REPLACE'? Sorry if this is a silly question but I have very little experience of databases.

Reply:
It does, yes. This is a MySQL-ism and not part of an SQL standard and we use it pretty frequently in the code.